From f9bc6b381bc6b7f73bf8807133fcc894d605a139 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Tue, 7 Oct 2025 22:08:46 +0300
Subject: [PATCH 353/553] tst_QHashFunctions: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: Ide2d11346a24427612a9142f4b3f999351de279d
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 305b5c51f1bfd841d007f2ededb4e87fa9efaecd)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../corelib/tools/qhashfunctions/tst_qhashfunctions.cpp   | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/auto/corelib/tools/qhashfunctions/tst_qhashfunctions.cpp b/tests/auto/corelib/tools/qhashfunctions/tst_qhashfunctions.cpp
index dec0f15ba39..5350d9e3f09 100644
--- a/tests/auto/corelib/tools/qhashfunctions/tst_qhashfunctions.cpp
+++ b/tests/auto/corelib/tools/qhashfunctions/tst_qhashfunctions.cpp
@@ -401,15 +401,15 @@ void tst_QHashFunctions::qhash()
     }
 
     {
-        QPair<int, int> p12(1, 2);
-        QPair<int, int> p21(2, 1);
+        std::pair<int, int> p12(1, 2);
+        std::pair<int, int> p21(2, 1);
 
         QVERIFY(qHash(p12, seed) == qHash(p12, seed));
         QVERIFY(qHash(p21, seed) == qHash(p21, seed));
         QVERIFY(qHash(p12, seed) != qHash(p21, seed));  // not guaranteed
 
-        QPair<int, int> pA(0x12345678, 0x12345678);
-        QPair<int, int> pB(0x12345675, 0x12345675);
+        std::pair<int, int> pA(0x12345678, 0x12345678);
+        std::pair<int, int> pB(0x12345675, 0x12345675);
 
         QVERIFY(qHash(pA, seed) != qHash(pB, seed));    // not guaranteed
     }
-- 
2.51.2

